torch/csrc/jit/passes/common_subexpression_elimination.cpp

```diff
@@ -107,18 +107,43 @@ struct EqualNodeCSE {
 
 // The function implements common subexpression elimination.
 // Since the nodes are visited in topological order, one pass is enough.
-void EliminateCommonSubexpression(Block * block) {
+void EliminateCommonSubexpression(Block * block,
+                                  std::function<Node*(Node*)> parent_lookup_fn) {
   std::unordered_set<Node*, HashNodeCSE, EqualNodeCSE> subexprs;
   for (auto it = block->nodes().begin(); it != block->nodes().end(); ++ it) {
     auto node = *it;
     if (node->kind() == prim::PythonOp
         || node->kind() == prim::Print
-        || node->blocks().size() > 0
        ) {
       // Do NOT have enough information to do CSE on these nodes.
       continue;
     }
 
+    if (node->blocks().size()) {
+      // Traverse sub-blocks.
+      for (auto block : node->blocks()) {
+        EliminateCommonSubexpression(block,
+        [&](Node *n) {
```

Comment:
nit: can you please indent this? otherwise it looks as if you were just continuing code on the same level, while it's really an argument to the call